src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php

```diff
@@ -0,0 +1,244 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShopBundle\Translation\Exception\UnsupportedLocaleException;
+use PrestaShopBundle\Translation\Extractor\LegacyModuleExtractorInterface;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Be able to retrieve information from legacy translation files
+ */
+class ExternalModuleLegacySystemProvider extends AbstractProvider implements UseDefaultCatalogueInterface, SearchProviderInterface
+{
+    /**
+     * @var SearchProviderInterface the module provider
+     */
+    private $moduleProvider;
+
+    /**
+     * @var LoaderInterface the translation loader from legacy files
+     */
+    private $legacyFileLoader;
+
+    /**
+     * @var LegacyModuleExtractorInterface the extractor
+     */
+    private $legacyModuleExtractor;
+
+    /**
+     * @var string the module name
+     */
+    private $moduleName;
+
+    /**
+     * @var string the domain name
+     */
+    protected $domain;
+
+    public function __construct(
+        SearchProviderInterface $moduleProvider,
+        LoaderInterface $databaseLoader,
+        $resourceDirectory,
+        LoaderInterface $legacyFileLoader,
+        LegacyModuleExtractorInterface $legacyModuleExtractor
+    ) {
+        $this->moduleProvider = $moduleProvider;
+        $this->legacyFileLoader = $legacyFileLoader;
+        $this->legacyModuleExtractor = $legacyModuleExtractor;
+
+        parent::__construct($databaseLoader, $resourceDirectory);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTranslationDomains()
+    {
+        return ['#^' . $this->getModuleDomain() . '*#i'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentifier()
+    {
+        return 'external_legacy_module';
+    }
+
+    /**
+     * is/should module name be part of the API?
```

Comment:
imho this needs its own interface (so we can check for the existence of the method the right way)